### DISCUSSION TITLE
Creating posts with image in the body (not attached)

### DISCUSSION THREAD

**Discussion by nigini** [tinyBlogPub/microblog.pub#3] [Ideas]
One of the things I miss the most in v2.0 is that I cannot add images to my longer-form posts (like Articles).

I found this [conversation on the original repo](https://todo.sr.ht/~tsileo/microblog.pub/125) that suggests I am not alone.

I am thinking about adding the ability to upload images, not as attachments, but as a resource that can be linked to the "Article" page. Should that be available on any post? 

I believe the response to that should be grounded on the ActivityPub protocol, as we don't want to break federation. For instance, I found [this conversation](https://socialhub.activitypub.rocks/t/article-vs-note-vs-page/4148/8) about how different implementations thread content differently. I will keep looking around about issues with adding images to the content of posts...

**Comment by nigini** [comment]
Here is a proposed specification that has been implemented by some major players: https://codeberg.org/fediverse/fep/src/branch/main/fep/e232/fep-e232.md

**Comment by chri2** [comment]
This could be part of some kind of draft function:

* draft some writing
* while doing so: upload an image
  * the upload would need the page to load again
  * saving the up to that point written content
  * putting the file into the list showing an URL to include it
* keep editing and include the URLs shown for the files in your md
* if an URL isn't used inside the note/article it is appended like in the actual version of microblog.pub

**Reply by nigini** [reply]
The part where you describe the "upload an image" is pretty much what I was thinking. But I believe that making it part of a draft function is a bit hacky ??? What about:

- we separate the "upload file" (I had just added issue #8 yesterday.) :)
- you could call the upload for the file as soon as they are picked
- we make the list of files available (with URL)
- and we add an "attach" checkbox for each file in the list ??? (I could be convinced, though, of your "attach by default" idea).

The good part is you could add this to the create, update, and draft pages... What do you think?

PS: I agree that "save draft" is likely a low priority.
